src/main/kotlin/no/nav/familie/ks/sak/integrasjon/økonomi/utbetalingsoppdrag/UtbetalingsoppdragGenerator.kt

```diff
@@ -57,8 +60,33 @@ class UtbetalingsoppdragGenerator {
                 ),
             forrigeAndeler = forrigeTilkjentYtelse?.tilAndelDataLongId(skalSendeOvergangsordningAndeler) ?: emptyList(),
             nyeAndeler = nyTilkjentYtelse.tilAndelDataLongId(skalSendeOvergangsordningAndeler),
-            sisteAndelPerKjede = sisteAndelPerKjede.mapValues { it.value.tilAndelDataLongId() },
+            sisteAndelPerKjede = sisteAndelPerKjedeMedOvergangsordningAndelTattIHensyn,
         )
+    }
+
+    private fun endreSisteAndelPerKjedeHvisOvergangsordningandel(
+        sisteAndelPerKjede: Map<IdentOgType, AndelTilkjentYtelse>,
+        forrigeTilkjentYtelse: TilkjentYtelse?,
+    ): Map<IdentOgType, AndelDataLongId> {
+        if (forrigeTilkjentYtelse == null || sisteAndelPerKjede.none { it.value.type == YtelseType.OVERGANGSORDNING }) {
+            return sisteAndelPerKjede.mapValues { it.value.tilAndelDataLongId() }
+        }
+
+        return sisteAndelPerKjede.mapValues { (identOgType, andelTilkjentYtelse) ->
+            if (andelTilkjentYtelse.type == YtelseType.OVERGANGSORDNING) {
+                val overgangsordningsAndelerForAktørSisteTilkjentYtelse =
+                    forrigeTilkjentYtelse
+                        .overgangsordningAndelerOgOrdinærForOvergangsordningUtbetalingsmånedTilAndelDataLongId()
+                        .filter { it.personIdent == identOgType.ident }
+                        .singleOrNull { it.fom == OVERGANGSORDNING_UTBETALINGSMÅNED && it.tom == OVERGANGSORDNING_UTBETALINGSMÅNED }
+                        ?: throw IllegalStateException("Greide ikke å utlede overgangsordning andel fra siste tilkjente ytelse for behandling ${andelTilkjentYtelse.behandlingId}")
```

Comment:
smakssak: Jeg foretrekker error("foo") i stedet for throw IllegalStateException("foo)

Reply:
Helt enig, godt poeng Stig ⭐